### DISCUSSION TITLE
Continuous build 5ad5a9f

### DISCUSSION THREAD

**Discussion by Terracotta-6** [artisan-roaster-scope/artisan#1290] [General]
1. Event 3 was incorrectly triggered.    
2. Incorrect event 5 annotation location.
![43da77b876bd8b52a2c24bf371b05cd](https://github.com/artisan-roaster-scope/artisan/assets/149906183/90459d1f-1017-48da-96e5-a68e948cc4fc)

**Comment by Terracotta-6** [comment]
[23-11-05_0522.alog.txt](https://github.com/artisan-roaster-scope/artisan/files/13257786/23-11-05_0522.alog.txt)

**Reply by MAKOMO** [reply]
Thanks for your report. It seems that quantifiers got confused here, but we would need your settings to be able to duplicate. fully understand and hopefully fix this one. Could you please export your artisan settings via menu Config >> Save Settings into an artisan-settings.aset file and drag-and-drop it here on a text field on GitHub, renamed to artisan-settings.aset.txt? Thanks!

**Reply by MAKOMO** [reply]
The mapping from the slider positions back to the extra devices 4 (Slider 01) and 5 (Slider 23) works fine here. The quantification of the extra device 1, channel 1 ("Slider 4 OT1蓝") to the event type 4 ("Slider 4") works fine too.

For the quantification of the extra device 1, channel 2 ("Slider 2 OT2红") I see some confusion. The begin and end are mapped to event type 1 ("Slider 1") and the middle is mapped to event type 3 ("Slider 3"). Most likely you have to event quantifiers defined for event type 1 and event type 3 which both are taking as source the extra device 1, channel 2 ("Slider 2 OT2红"). Sometimes the one quantifier wins, sometimes the other. Note that per sample interval randomly only one active quantifier fires its action.

For now I do not see an issue in the Artisan software here, but a confusion in the configuration. Thus sharing your configuration and maybe giving us an idea what you intend to achieve can help us to clarify this one.

**Reply by Terracotta-6** [reply]
[artisan-settings.txt](https://github.com/artisan-roaster-scope/artisan/files/13259532/artisan-settings.txt)

**Comment by Terracotta-6** [comment]
Event 3 was incorrectly triggered Occurs only in this version. he previously released test version is normal.

**Reply by MAKOMO** [reply]
Sorry, I am still struggling to understand this whole thing. What is triggering "Event 3" (an alarm rule, a quantifier or an event playback or anything else) and which event is "Event 3" (the 3rd event in the event table)?

It seems you are replaying events from a background profile here. Could you please also share that background profile?

**Reply by MAKOMO** [reply]
Can you please provide me with detailed steps from a Factory Reset to duplicate the issue you are talking about? I am struggling.

**Reply by Terracotta-6** [reply]
Yes, following your advice, nothing has changed.

**Comment by Terracotta-6** [comment]
[profile background.txt](https://github.com/artisan-roaster-scope/artisan/files/13260819/profile.background.txt)

**Reply by Terracotta-6** [reply]
No alarm triggering events were used.

**Reply by MAKOMO** [reply]
I still do not understand your issue. However, I reworked some things. Could you please re-test with the current continuous build?

https://github.com/artisan-roaster-scope/artisan/releases/tag/continuous

Thank you!

**Reply by Terracotta-6** [reply]
![81eb9e14133f0279143b42177cbc296](https://github.com/artisan-roaster-scope/artisan/assets/149906183/20ee5fbe-2e6f-4fd4-b4e7-ee8ebeb8638a)

**Reply by Terracotta-6** [reply]
the 3rd event in the event table Event 3 ,Still being triggered incorrectly.

**Comment by Terracotta-6** [comment]
This version started having problems.
![3732c8ee7aebc664e985faf21221c15](https://github.com/artisan-roaster-scope/artisan/assets/149906183/9816bb9b-6f0e-436c-b742-5a63c79b42e2)

**Comment by Terracotta-6** [comment]
![cea6423f918132a675d6b7113a62f5a](https://github.com/artisan-roaster-scope/artisan/assets/149906183/c8c18609-1db6-4094-b9b1-29b934bf80fc)
Correct test version.

**Comment by MAKOMO** [comment]
I still understand where the problem is and don't understand what your refer to by "the 3rd event in the event table Event 3". I also do not have an TC4 to test.

I load your settings, your background profile and your foreground profile that you attached here above. This looks like this to me:

<img width="1273" alt="Screenshot 2023-11-10 at 08 58 49" src="https://github.com/artisan-roaster-scope/artisan/assets/13909011/481e19d6-8d65-45b6-b0e1-48c147f021a2">

I see that you do not have any event quantifier defined and all alarms are turned off. However, you have the background event replay by time turned on.

<img width="922" alt="Screenshot 2023-11-10 at 09 01 07" src="https://github.com/artisan-roaster-scope/artisan/assets/13909011/71a2e5ef-9dd4-4153-ae37-d513f7efaac5">

The events to be replayed by the background profiles event table are the followings with the 3rd event marked (is it this one you are talking about!?):

<img width="922" alt="Screenshot 2023-11-10 at 09 02 55" src="https://github.com/artisan-roaster-scope/artisan/assets/13909011/f8791ce7-23d5-4a99-a2f1-2a58dab66d05">

Some additional "untyped" events are marked as "error" or "correct". Those are of course not replayed as they do not have one of the four event types assigned. Not sure what this means.

No looking at the event table of your foreground profile I see all those events from the background exactly replayed (just that at charge 4 additional events are generated to mark the state at CHARGE). What is wrong here?

<img width="815" alt="Screenshot 2023-11-10 at 09 04 23" src="https://github.com/artisan-roaster-scope/artisan/assets/13909011/71b49e02-418b-45f6-9ec5-b2157f4f4a69">

00:00 Slider 1 => 5
00:00 Slider 2 => 20
00:00 Slider 3 => 44
00:00 Slider 4 => 23
01:15 Slider 2 => 30
01:25 Slider 4 => 47
02:00 Slider 3 => 55
02:15 Slider 1 => 10
...

I cannot see anything wrong with the event replay.

Also your extra devices 4 and 5 recording "Slider 01" and "Slider 23" movement are spot on with the background profiles event and the replay of the foreground events.

What I see is that the second channel of your first extra device channel named "Slider 2 OT2红", recording the "ArduinoTC4" channel 6 is first following the slider 3 signal and after minute 2:00 slider 1.  This is easily explained as your slider 3 sends out an "IO3;{}" command which is just a variant of the "DCFAN;{}" command in the TC4 PID firmware, modifying the same TC4 variable (intended to regulate the fan speed).

In your "Correct test version." screen shot those lines are not drawn at all. Why?

So I still don't see what is wrong in the current version and what is different to previous builds.

What event is exactly triggered wrong?

Maybe you can simplify the example and remove things that do not contribute like all those extra device lines?

Sorry, I really would love to fix what ever the issue is here, but as long as I don't understand the issue nor can replicate it, I will not able to do so.

Thanks for your support!

**Comment by Terracotta-6** [comment]
This beta version is working fine again! Since you don't have the physical hardware Hardware-In-the-Loop, thank you very much for taking your precious time to answer！
![e63526c42686e2622f0e07c318b2688](https://github.com/artisan-roaster-scope/artisan/assets/149906183/24fcd29f-b465-477a-ad5d-5aeb4795b863)

**Reply by MAKOMO** [reply]
Thank you for testing and this confirmation that the current builds are working for you again. We plan to make a new Artisan release by the end of this month thus we want everything to work at least as good as the previous version (hopefully even a bit better ;) after all the many changes we applied.
